presto-spi/src/main/java/com/facebook/presto/spi/plan/OutputNode.java

```diff
@@ -61,17 +58,19 @@ public OutputNode(
 
         requireNonNull(source, "source is null");
         requireNonNull(columnNames, "columnNames is null");
-        Preconditions.checkArgument(columnNames.size() == outputVariables.size(), "columnNames and assignments sizes don't match");
+        if (columnNames.size() != outputVariables.size()) {
+            throw new IllegalArgumentException("columnNames and assignments sizes don't match");
+        }
 
         this.source = source;
         this.columnNames = columnNames;
-        this.outputVariables = ImmutableList.copyOf(outputVariables);
+        this.outputVariables = unmodifiableList(outputVariables);
```

Comment:
why did this change from Immutable to unmodifiable?

Reply:
It's also a good idea to replace Guava and similar libraries with JDK methods where available. A lot of Guava, Apache commons collections, and the like exist primarily to provide utilities that are available in JDK 8+ for earlier JDK versions.